spec/support/evm_spec_helper.rb

```diff
@@ -46,7 +46,37 @@ def self.clear_caches
     # Clear the thread local variable to prevent test contamination
     User.current_user = nil if defined?(User) && User.respond_to?(:current_user=)
 
-    ::Settings.reload! if Digest::MD5.hexdigest(::Settings.to_json) != settings_digest
+    settings_restore if settings_changed?
+  end
+
+  private_class_method def self.settings_backup
+    @settings_backup ||= settings_dump
+  end
+
+  private_class_method def self.settings_restore
+    Vmdb::Settings.reset_settings_constant(settings_load(@settings_backup))
+  end
+
+  private_class_method def self.settings_dump
+    Marshal.dump(::Settings).tap do |dump|
+      # Marshal dump of Settings loses the config_sources, so we need to backup manually
+      dump.instance_variable_set(:@config_sources, ::Settings.instance_variable_get(:@config_sources).dup)
+    end
+  end
+
+  private_class_method def self.settings_load(dump)
+    Marshal.load(dump).tap do |settings|
+      # Marshal dump of Settings loses the config_sources, so we need to restore manually
+      settings.instance_variable_set(:@config_sources, dump.instance_variable_get(:@config_sources).dup)
```

Comment:
Is this the only ivar OR only one lost?  It's curious that they don't marshaled.

Reply:
There are 2 ivars, `@table` (from OpenStruct) and `@config_sources`. however, OpenStruct has defined `marshal_dump` as only returning `@table`.  So, this adds a custom backup of that one.